### PR TITLE
Option to maintain local cache file in Remote

### PR DESCRIPTION
**Pull request by arumugaraja-thirumani**
With `remote_file_list_cache_file_path` options we could be able to reduce and sync time by 30 minutes.

Maintaining the local file is difficult in the docker environment, so in this PR providing the option to have the cache file in remote (`remote_file_list_remote_path`) as well. Which will be downloaded during asset sync if path is configured and its creates a local file for further references in the sync. Once after the sync is completed, updated local cache file will be uploaded back to configured remote path. 

This remote cache file will be maintained in the path configured using `remote_file_list_remote_path` inside `fog_directory`

### CODE REVIEW

File: lib/asset_sync/storage.rb [hunk 1]
```diff
@@ -317,6 +342,7 @@ def upload_files
       end
 
       update_remote_file_list_cache(local_files_to_upload)
+      update_remote_file_list_in_remote if remote_file_list_remote_path
```
Comment (PikachuEXE) [inline lib/asset_sync/storage.rb:345]: Why not also put `if remote_file_list_remote_path` into `update_remote_file_list_in_remote`?

File: lib/generators/asset_sync/templates/asset_sync.rb [hunk 2]
```diff
@@ -81,6 +81,9 @@
     # Path to cache file to skip scanning remote
     # config.remote_file_list_cache_file_path = './.asset_sync_remote_file_list_cache.json'
     #
+    # Path to maintain remote file list in remote
```
Comment (PikachuEXE) [inline lib/generators/asset_sync/templates/asset_sync.rb:84]: I think we need better explanation here...
Something like `Path on remote storage to persist remote file list file`?

Already I am confused here too

File: README.md [hunk 3]
```diff
@@ -380,6 +383,7 @@ AssetSync.config.gzip_compression == ENV['ASSET_SYNC_GZIP_COMPRESSION']
 * **concurrent_uploads**: (`true, false`) when enabled, will upload the files in different Threads, this greatly improves the upload speed. **default:** `'false'`
 * **concurrent_uploads_max_threads**: when concurrent_uploads is enabled, this determines the number of threads that will be created. **default:** `10`
 * **remote_file_list_cache_file_path**: if present, use this path to cache remote file list to skip scanning remote **default:** `nil`
+* **remote_file_list_remote_path**: if present, use this path to download remote file list file to cache file list in local to skip scanning remote. useful in container environment where you cannot maintain the local file **default:** `nil`
```
Comment (PikachuEXE) [inline README.md:386]: Doesn't mention `remote_file_list_cache_file_path` is needed

**Review by PikachuEXE — COMMENTED** [review]
I think the feature is valid but the documentation probably need some improvements

File: lib/asset_sync/storage.rb (hunk 1 above)
Comment (arumugaraja-thirumani) [reply lib/asset_sync/storage.rb]: Updated

File: README.md (hunk 3 above)
Comment (arumugaraja-thirumani) [reply README.md]: Updated

File: lib/generators/asset_sync/templates/asset_sync.rb (hunk 2 above)
Comment (arumugaraja-thirumani) [reply lib/generators/asset_sync/templates/asset_sync.rb]: Updated the suggestion

**Review by PikachuEXE — APPROVED** [review]

**Comment by PikachuEXE** [comment]
Released in `2.16.0`

**Pull request merged**